.github/workflows/ci.yml

```diff
@@ -579,6 +579,64 @@ jobs:
+      - cpu
+      - os-family=Linux
+    env:
+      BUILD_DIR: ${{ needs.build_all.outputs.build-dir }}
+      BUILD_DIR_ARCHIVE: ${{ needs.build_all.outputs.build-dir-archive }}
+      BUILD_DIR_GCS_ARTIFACT: ${{ needs.build_all.outputs.build-dir-gcs-artifact }}
+      TF_BINARIES_DIR: ${{ needs.build_tf_integrations.outputs.binaries-dir }}
+      TF_BINARIES_ARCHIVE: ${{ needs.build_tf_integrations.outputs.binaries-archive }}
+      TF_BINARIES_GCS_ARTIFACT: ${{ needs.build_tf_integrations.outputs.binaries-gcs-artifact }}
+    outputs:
+      e2e-test-artifacts-dir: ${{ steps.build.outputs.e2e-test-artifacts-dir }}
+      e2e-test-artifacts-gcs-artifact-dir: ${{ steps.upload.outputs.e2e-test-artifacts-gcs-artifact-dir }}
+    steps:
+      - name: "Checking out repository"
+        uses: actions/checkout@7884fcad6b5d53d10323aee724dc68d8b9096a2e # v2
+      - name: "Checking out runtime submodules"
+        run: ./build_tools/scripts/git/update_runtime_submodules.sh
+      - name: "Downloading build dir archive"
+        run: gcloud alpha storage cp "${BUILD_DIR_GCS_ARTIFACT}" "${BUILD_DIR_ARCHIVE}"
+      - name: "Extracting install from build dir archive"
+        run: tar -xf "${BUILD_DIR_ARCHIVE}" "${BUILD_DIR}/install"
+      - name: "Downloading TF binaries archive"
+        run: gcloud alpha storage cp "${TF_BINARIES_GCS_ARTIFACT}" "${TF_BINARIES_ARCHIVE}"
+      - name: "Extracting TF binaries archive"
+        run: tar -xf "${TF_BINARIES_ARCHIVE}"
+      - name: "Building e2e test artifacts"
+        id: build
+        env:
+          BUILD_E2E_TEST_ARTIFACTS_DIR: build-e2e-test-artifacts
+        run: |
+          build_tools/github_actions/docker_run.sh \
+            --env "IREE_TF_BINARIES_DIR=${TF_BINARIES_DIR}" \
+            --env "IREE_HOST_BINARY_ROOT=${BUILD_DIR}/install" \
+            --env "BUILD_E2E_TEST_ARTIFACTS_DIR=${BUILD_E2E_TEST_ARTIFACTS_DIR}" \
+            gcr.io/iree-oss/base@sha256:7c3027c48b94fc38e64488987fc7893c100526c57308d25cef0c6b76a2dfe117 \
+            build_tools/cmake/build_e2e_test_artifacts.sh
+          echo "::set-output name=e2e-test-artifacts-dir::${BUILD_E2E_TEST_ARTIFACTS_DIR}/e2e_test_artifacts"
+      - name: "Uploading e2e test artifacts"
+        id: upload
+        env:
+          E2E_TEST_ARTIFACTS_DIR: ${{ steps.build.outputs.e2e-test-artifacts-dir }}
+          E2E_TEST_ARTIFACTS_GCS_ARTIFACT_DIR: ${{ env.GCS_DIR }}/e2e-test-artifacts
+        run: |
+          # Not archiving the directory to allow fetching each file as needed
+          # separately. The option `--gzip-local-all` compresses files first
+          # and upload them to the GCS in the gzip format. The files can be
+          # either on-the-fly or manually uncompressed when being downloaded.
+          # More details: https://cloud.google.com/storage/docs/transcoding
+          gcloud alpha storage cp -r --gzip-local-all "${E2E_TEST_ARTIFACTS_DIR}/iree" "${E2E_TEST_ARTIFACTS_GCS_ARTIFACT_DIR}/iree"
```

Comment:
Neat. Did you weigh this against `gzip-in-flight`? Seems like the difference is whether the object is stored compressed or not. The caller might prefer the uncompressed file on download, since they're likely to have higher bandwidth. Not a big deal either way though

Maybe also link to https://cloud.google.com/sdk/gcloud/reference/storage/cp

Reply:
I'm thinking it will save the storage if we store files in the compressed format. Downloading those files will be in uncompressed format by default, unless users specify `Accept-Encoding: gzip` in the download request.

If saving storage is not a concern, I also prefer to use `gzip-in-flight`, just simpler semantic.

Reply:
> Downloading those files will be in uncompressed format by default, unless users specify Accept-Encoding: gzip in the download request.

Hmm that's not my understanding of the docs:

> However, the Content-Encoding metadata is set to gzip and the Cache-Control metadata set to no-transform. The data remains compressed on Cloud Storage servers and will not be decompressed on download by gcloud stroage because of the no-transform field.

A tleast, Depends on how much

Reply:
Oh, I missed that section. But somehow when I was testing, the file was downloaded in an uncompressed format, both with gcloud and wget. For example:
```
gcloud alpha --project iree-oss storage cp gs://iree-github-actions-presubmit-artifacts/3527399884/1/e2e-test-artifacts/iree/0e466f69-91d6-4e50-b62b-a82b6213a231_MobileSSD_fp32/MobileSSD_fp32.mlir
```

In some cases the VMFB is compressed well. For example the MiniLM module is 133MB uncompressed and 78MB when compressed.

I'll say we just use `gzip-in-flight` first to avoid the unclear behavior on the download.

Reply:
Sounds good. Thanks for doing the investigation